File: Лаб/Лаб1/лаб 1-2/TwoSum.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (numMap.containsKey(complement)) {
                return new int[] { numMap.get(complement), i };
            }

            numMap.put(nums[i], i);
        }

        return new int[] {};
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        try {
            BufferedReader reader = new BufferedReader(new FileReader("input1.txt"));
            String[] numStrs = reader.readLine().split(" ");
            int[] nums = new int[numStrs.length];

            for (int i = 0; i < numStrs.length; i++) {
                nums[i] = Integer.parseInt(numStrs[i]);
            }

            int target = Integer.parseInt(reader.readLine());
            reader.close();

            int[] result = solution.twoSum(nums, target);

            BufferedWriter writer = new BufferedWriter(new FileWriter("output1.txt"));
            if (result.length == 2) {
                writer.write(result[0] + " " + result[1]);
            } else {
                writer.write("Hariult baihgui baina");
            }
            writer.close();

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Лаб/Лаб1/Twosum.py
class Solution:
    def twoSum(self, nums, target):
        num_map = {}
        for i, num in enumerate(nums):
            complement = target - num
            
            if complement in num_map:
                return [num_map[complement], i]
            
            num_map[num] = i
        return []
def main():
    try:
        with open("input1.txt", "r") as file:
            nums = list(map(int, file.readline().split()))
            target = int(file.readline())
        solution = Solution()
        result = solution.twoSum(nums, target)
        with open("output1.txt", "w") as file:
            if result:
                file.write("{} {}".format(result[0], result[1]))
            else:
                file.write("Hariult baihgui baina")
    
    except IOError as e:
        print("File-iig unshih esvel bichihed aldaa garlaa {}".format(e))

if __name__ == "__main__":
    main()
